eng/common/scripts/Service-Level-Readme-Automation.ps1

```diff
@@ -42,51 +42,64 @@ param(
+      # The existing service level readme contains both metadata and index table
+      if ($readmeContent -match "^---(\r?\n.*){1,30}\[\!INCLUDE\s\[.*-packages\]\(.*-index.md\)\](?<content>(\r?\n.*)*)$") {
+        $restContent = $Matches["content"].trim()
+      }
+    }
+    elseif ($readmeContent -match "^---") {
+      # The existing service level readme contains only metadata
+      if ($readmeContent -match "^---\n*(.*\n?)*?---\n*(?<content>(.*\n?)*)") {
+        $restContent = $Matches["content"].trim()
+      }
+    }
+    else {
+      # Readme contains neither metadata nor index table.
+      $restContent = $readmeContent
+    }
+  }
+  return $restContent
+}
+
+# Always update metadata and index table.
+function update-service-readme($readmeFolder, $readmeName, $moniker, $msService, $clientTableLink, $mgmtTableLink, $serviceName)
 {
-  $readmePath = Join-Path $readmeFolder -ChildPath $readmeName
-  $content = ""  
+  $readmePath = (Join-Path $readmeFolder -ChildPath $readmeName)
+  $restContent = Get-RestContent -readmePath $readmePath
+  $content = ""
   if (Test-Path (Join-Path $readmeFolder -ChildPath $clientTableLink)) {
-    $content = "## Client packages - $moniker`r`n"
-    $content += "[!INCLUDE [client-packages]($clientTableLink)]`r`n"
+    $content += "## Client packages - $moniker`n"
+    $content += "[!INCLUDE [client-packages]($clientTableLink)]`n"
   }
   if (Test-Path (Join-Path $readmeFolder -ChildPath $mgmtTableLink)) {
-    $content += "## Management packages - $moniker`r`n"
-    $content += "[!INCLUDE [mgmt-packages]($mgmtTableLink)]`r`n"
+    $content += "## Management packages - $moniker`n"
+    $content += "[!INCLUDE [mgmt-packages]($mgmtTableLink)]"
   }
   if (!$content) {
-    return
+    if (Test-Path $readmePath) {
+      Remove-Item $readmePath -Force
+    }
   }
   # Generate the front-matter for docs needs
   # $Language, $LanguageDisplayName are the variables globally defined in Language-Settings.ps1
   $metadataString = GenerateDocsMsMetadata -language $Language -languageDisplayName $LanguageDisplayName -serviceName $serviceName `
     -tenantId $TenantId -clientId $ClientId -clientSecret $ClientSecret `
     -msService $msService
-  Add-Content -Path $readmePath -Value $metadataString -NoNewline
+  Set-Content -Path $readmePath -Value $metadataString -NoNewline
 
   # Add tables, seperate client and mgmt.
-  $readmeHeader = "# Azure $serviceName SDK for $languageDisplayName - $moniker`r`n"
-  Add-Content -Path $readmePath -Value $readmeHeader
-  Add-Content -Path $readmePath -Value $content -NoNewline
-}
-
-# Update the metadata table.
-function update-metadata-table($readmeFolder, $readmeName, $serviceName, $msService)
-{
-  $readmePath = Join-Path $readmeFolder -ChildPath $readmeName
-  $readmeContent = Get-Content -Path $readmePath -Raw
-  $match = $readmeContent -match "^---\n*(?<metadata>(.*\n?)*?)---\n*(?<content>(.*\n?)*)"
```

Comment:
Based on what I'm seeing you are trying to now exclude the table with the links from the content here. Why do we need to do that? I thought that was exactly why we split the package table into another MD that folks could include if they want. If they don't do the include then they won't get the table.